nuclio/config.py

```diff
@@ -203,15 +203,49 @@ def set_env(config, env):
             raise ValueError(
                 'cannot parse environment value from: {}'.format(line))
 
+        # TODO: change to allow magic secrets
         update_env_var(config, key, value)
 
 
 def set_env_dict(config, env={}):
     for k, v in env.items():
-        update_env_var(config, k, str(v))
+        update_env_var(config, k, value=str(v))
+
+
+def set_secrets_dict(config, secrets={}):
+    for k, v in secrets.items():
+        try:
+            v_dict = json.loads(v)
+        except ValueError as e:
+            logger.warning(f'failed to deserialize env variable from secret: {v}')
+            raise e
+
+        secret_key_ref = v_dict.get('secretKeyRef', {}) or v_dict.get('secret_key_ref', {})
+        name = secret_key_ref.get('name', '')
+        secret_key = secret_key_ref.get('key', '')
+        if not name or not secret_key:
+            logger.info(f'skipping nameless or keyless env variable from secret: {v}')
```

Comment:
is that a desire or a valid flow?
if not, raise an exception, let the user know his input is invalid